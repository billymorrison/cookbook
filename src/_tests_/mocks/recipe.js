export const recipe = {
    _id: "5ff4bcbdc91e0e20e2a5be26",
    title: "Mexican chicken burger",
    displayName: "tomo",
    prepTime: "15 mins",
    cookTime: "60 mins",
    difficulty: "Hard",
    serves: 4,
    ingredients: "onion",
    method:
      "In a large, heavy saucepan, heat the oil. Add the onion, garlic, green chilli, ginger and some seasoning. Fry on a medium heat for 10 mins or until soft",
    __v: 0,
}