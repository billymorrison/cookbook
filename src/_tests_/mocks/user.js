export const user = {
    _id: 123,
    displayName: "billy",
};