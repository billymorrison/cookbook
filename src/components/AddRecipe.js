import {
  StyledForm,
  StyledLabel,
  StyledInput,
  StyledButton,
  MainLoginArea,
} from "./GlobalStyles";

import styled from "styled-components";

const DoubleInput = styled.div`
  display: flex;
  justify-content: space-between;
`;
const IngredientInput = styled(StyledInput)`
  max-width: 49%;
`;

const AddRecipe = ({ handleChange }) => {
  return (
    <>
      <StyledForm>
        <StyledLabel>
          Title
          <StyledInput
            type="text"
            required
            onChange={handleChange}
            name="title"
          />
        </StyledLabel>
        <StyledLabel>
          Cook Time
          <StyledInput type="text" onChange={handleChange} name="cookTime" />
        </StyledLabel>
        <StyledLabel>
          Difficulty
          <StyledInput
            type="text"
            required
            onChange={handleChange}
            name="difficulty"
          />
        </StyledLabel>
        <StyledLabel>
          Serves
          <StyledInput
            type="number"
            required
            onChange={handleChange}
            name="serves"
          />
        </StyledLabel>
        <StyledLabel>
          Prep Time
          <StyledInput
            type="text"
            required
            onChange={handleChange}
            name="prepTime"
          />
        </StyledLabel>
        <StyledLabel>
          Ingredients
          <DoubleInput>
            <StyledLabel>
              Ingredient
              <IngredientInput
                type="text"
                required
                onChange={handleChange}
                name="ingredientKey"
              />
            </StyledLabel>
            <StyledLabel>
              Quantity
              <IngredientInput
                type="text"
                required
                onChange={handleChange}
                name="ingredientValue"
              />
            </StyledLabel>
          </DoubleInput>
        </StyledLabel>
        <StyledLabel>
          Method
          <StyledInput
            type="text"
            required
            onChange={handleChange}
            name="method"
          />
        </StyledLabel>
        <StyledButton />
      </StyledForm>
    </>
  );
};

export default AddRecipe;
