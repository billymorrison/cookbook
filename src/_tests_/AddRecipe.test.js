import ReactDOM from "react-dom";
import AddRecipe from "../components/AddRecipe";
import { user } from "./mocks/user";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders add recipe component without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<AddRecipe user={user}></AddRecipe>, div)
});

it("renders add recipe correctly", () => {
  const {getByTestId} = render(<AddRecipe user={user}></AddRecipe>);
  expect(getByTestId("addRecipe")).toHaveTextContent("Title");
})

it("matches snapshot", () => {
  const tree = renderer.create(<AddRecipe user={user}></AddRecipe>).toJSON();
  expect(tree).toMatchSnapshot();
})