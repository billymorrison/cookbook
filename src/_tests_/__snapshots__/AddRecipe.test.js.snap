// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="sc-hKgILt sc-eCssSg icauyc jACgWr"
  data-testid="addRecipe"
  encType="multipart/form-data"
  onSubmit={[Function]}
>
  <label
    className="sc-gsTCUz fYmKmH"
  >
    Title
    <input
      className="sc-bdfBwQ bDhmHQ"
      name="title"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </label>
  <div
    className="sc-pFZIQ hDlVFw"
  >
    <label
      className="sc-gsTCUz sc-jrAGrp fYmKmH gnLDqD"
    >
      Prep Time
      <input
        className="sc-bdfBwQ sc-kEjbxe bDhmHQ ipVbvI"
        name="prepTime"
        onChange={[Function]}
        required={true}
        type="text"
      />
    </label>
    <label
      className="sc-gsTCUz sc-jrAGrp fYmKmH gnLDqD"
    >
      Cook Time
      <input
        className="sc-bdfBwQ sc-kEjbxe bDhmHQ ipVbvI"
        name="cookTime"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
  </div>
  <div
    className="sc-pFZIQ hDlVFw"
  >
    <label
      className="sc-gsTCUz sc-jrAGrp fYmKmH gnLDqD"
    >
      Difficulty
      <select
        className="sc-iqHYGH gLrfSs"
        name="difficulty"
        onChange={[Function]}
        value="Easy"
      >
        <option
          value="easy"
        >
          Easy
        </option>
        <option
          value="medium"
        >
          Medium
        </option>
        <option
          value="hard"
        >
          Hard
        </option>
      </select>
    </label>
    <label
      className="sc-gsTCUz sc-jrAGrp fYmKmH gnLDqD"
    >
      Serves
      <input
        className="sc-bdfBwQ sc-kEjbxe bDhmHQ ipVbvI"
        name="serves"
        onChange={[Function]}
        required={true}
        type="number"
        value=""
      />
    </label>
  </div>
  <label
    className="sc-gsTCUz fYmKmH"
  >
    Ingredients
    <textarea
      className="sc-fubCfw duJyEv"
      name="ingredients"
      onChange={[Function]}
      required={true}
      type="text"
    />
  </label>
  <label
    className="sc-gsTCUz fYmKmH"
  >
    Method
    <textarea
      className="sc-fubCfw duJyEv"
      name="method"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <label
      className="sc-gsTCUz fYmKmH"
    >
      Image
      <input
        className="sc-bdfBwQ bDhmHQ"
        name="image"
        onChange={[Function]}
        required={true}
        type="file"
      />
    </label>
  </label>
  
  <button
    className="sc-jSgupP iSLeyX"
  >
    Submit
  </button>
</form>
`;
